example/main.go

```diff
@@ -29,7 +29,9 @@ type DatabaseConfig struct {
 
 type Config struct {
 	Debug    bool           `env:"DEBUG"`
+	Host     string         `env:"HOST,default=localhost"`
 	Port     string         `env:"PORT,default=8080"`
+	Address  string         `env:"ADDRESS,default=${HOST}:$PORT,expand"`
```

Comment:
You should use different name here than host and port, because the example might let people think HOST and PORT already need to be provided as fields

So maybe, something like


```suggestion
	Address  string         `env:"ADDRESS,default=${ADDR}:$BIND,expand"`
```

Reply:
good point about the default value for variable used in another variable using expand.

It should be covered by a dedicated test maybe

Setting just

PORT="8081"

and expect to have

```go
type Config struct {
	Debug    bool           `env:"DEBUG"`
	Host     string         `env:"HOST,default=localhost"`
	Port     string         `env:"PORT,default=8080"`
	Address  string         `env:"ADDRESS,default=${HOST}:$PORT,expand"`
}
```

```
cfg := Config{
	Host: "localhost"
	Port: "8081"
	Address: "localhost:8081"
}
```